backend/substrapp/views/aggregatetuple.py

```diff
@@ -23,17 +25,24 @@ def get_queryset(self):
     def perform_create(self, serializer):
         return serializer.save()
 
-    def commit(self, serializer, channel_name, pkhash):
+    def commit(self, serializer, channel_name):
         # create on ledger
         try:
-            data = serializer.create(channel_name, serializer.validated_data)
+            data = serializer.validated_data
+            if data['rank'] == 0 and not data['compute_plan_id']:
+                # Auto-create compute plan
+                res = create_compute_plan(channel_name, data={})
+                data['compute_plan_id'] = res['compute_plan_id']
+            data = serializer.create(channel_name, data)
```

Comment:
I think the serializer should take care only of its own type (Aggregate Tuple in this case), and not manipulate other types (Compute Plan). I'd argue that's the responsibility of the view.

> Is it a behavior we want to have ?

I simply kept on supporting this use-case (create a compute plan automatically) :) It was implemented in the chaincode prior to this PR but now it's implemented in the backend (because the backend generates keys). We could consider dropping this feature in the future cc @inalgnu